client/subchannel_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2022 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client_test
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	ethchanneltest "github.com/perun-network/perun-eth-backend/channel/test"
+	ethclienttest "github.com/perun-network/perun-eth-backend/client/test"
+	"perun.network/go-perun/apps/payment"
+	chtest "perun.network/go-perun/channel/test"
+	"perun.network/go-perun/client"
+	clienttest "perun.network/go-perun/client/test"
+	"perun.network/go-perun/wire"
+	pkgtest "polycry.pt/poly-go/test"
+)
+
+func TestSubChannelHappy(t *testing.T) {
+	rng := pkgtest.Prng(t)
+	const A, B = 0, 1 // Indices of clients.
+	s := ethchanneltest.NewSetup(t, rng, 2, ethclienttest.BlockInterval, TxFinalityDepth)
+	setups := ethclienttest.MakeRoleSetups(rng, s, [2]string{"Susie", "Tim"})
+	roles := [2]clienttest.Executer{
+		clienttest.NewSusie(t, setups[A]),
+		clienttest.NewTim(t, setups[B]),
+	}
+	// Enable stage synchronization.
+	stages := roles[A].EnableStages()
+	roles[B].SetStages(stages)
+
+	// Build configuration.
+	baseCfg := clienttest.MakeBaseExecConfig(
+		[2]wire.Address{setups[A].Identity.Address(), setups[B].Identity.Address()},
+		s.Asset,
+		[2]*big.Int{big.NewInt(100), big.NewInt(100)},
+		client.WithoutApp(),
+	)
+	const (
```

Comment:
Looks more readable if the constants and variables are defined in the beginning of the method.

Reply:
I think it's a matter of taste. Also, this part did not change since you approved last time.